cpp/csp/adapters/websockets/ClientHeaderUpdateAdapter.cpp

```diff
@@ -0,0 +1,26 @@
+#include <csp/adapters/websockets/ClientHeaderUpdateAdapter.h>
+
+namespace csp::adapters::websockets {
+
+ClientHeaderUpdateAdapter::ClientHeaderUpdateAdapter(
+    Engine * engine,
+    Dictionary& properties
+) : OutputAdapter( engine ), m_properties(properties)
+{
+
+};
+
+void ClientHeaderUpdateAdapter::executeImpl()
+{
+    std::vector<std::string> value = input() -> lastValueTyped<std::vector<std::string>>();
+    DictionaryPtr headers = m_properties.get<DictionaryPtr>("headers");
+    // this is hacky but it works...
+    for( size_t i = 0; i < value.size(); i+=2)
+    {
+        const std::string key = value[i];
+        const std::string v= value[i+1];
+        headers->update(key, v);
```

Comment:
do you want to support clearing a header too?
if you do make it a struct you can take an unset value as a clear

Reply:
I'm not sure dictionaries have a clear in the cpp impl. https://github.com/Point72/csp/blob/main/cpp/csp/engine/Dictionary.h#L25

Reply:
They do not, thats why I said "if you do make it a struct"

Reply:
I think for now, I'm going to hold off on clearing a header or making the headers a struct themselves. If there is interest down the line it is something to look into supporting.